server/models.py

```diff
@@ -266,9 +266,10 @@ def can(cls, obj, user, action):
             return False
         if user.is_admin:
             return True
+        is_staff = user.is_enrolled(obj.course.id, STAFF_ROLES)
         if action == "view":
-            return user.is_authenticated
-        return user.is_enrolled(obj.course.id, STAFF_ROLES)
+            return is_staff or obj.visible
```

Comment:
The student endpoint doesn't do a `.can` check, so this won't affect the behavior of the assignment page on the student dashboard.